source/components/NumericInput.js

```diff
@@ -334,6 +334,7 @@ class NumericInput extends Component<Props, State> {
 
   _isNumeric(value: string) {
     const replacedValue = value.replace(/,/g, '');
+    // eslint-disable-next-line no-restricted-globals
     return !isNaN(parseFloat(replacedValue)) && isFinite(replacedValue);
```

Comment:
@nikolaglumac 
This is exactly the line of code that threw me for a big loop yesterday. When I was initially implementing ESLint, it threw an error here asking me to change **!isNaN** to **!Number.isNaN**. I made this change per ESLint's documentation because it was related to browser support. It was misleading because this isn't about browser support at all! My second mistake was that I didn't open storybook and test NumericInput at that time. I wrongly assumed the changes I made for liniting were mostly stylistic or related to checking for non-existent properties on an object at the top level of a for...in loop. Either way, part of our QA is running the code again after any significant changes such as these. I did however test NumericInput yesterday and spent a lot of time looking for this bug, finally found it and corrected it, but did not disable the line. Thank you for adding this. I've looked into this issue more and it is quite a rabbit hole, but I am going to think about more because we are assuming **_isNumeric** always receives a string. I am not sure if that is a safe assumption either.

Reply:
@MarcusHurney I am pretty confident that we do receive a string there.
I believe it makes sense to investigate this part of the code as NumericInput is a very important component in Daedalus. For example this part: `parseFloat(replacedValue)` will convert value such as `1abc.000000` into `1` 😱 
Please invest more time in this plus in adding `// @flow` declarations in all of the JS files.
Thanks!

Reply:
@nikolaglumac 
Will do! I'm adding the // @flow declarations for the remaining files, and will continue researching the best approach for the _isNumeric method. I was also just thinking, man this is a pretty important component... 😅

Reply:
Great :)